src/erlang_ls_client.erl

```diff
@@ -268,59 +264,54 @@ handle_call({initialize, RootUri, InitOptions}, From, State) ->
                   }
              },
   Content = erlang_ls_protocol:request(RequestId, Method, Params),
-  gen_tcp:send(Socket, Content),
+  send(Content, State),
   {noreply, State#state{ request_id = RequestId + 1
                        , pending    = [{RequestId, From} | State#state.pending]
                        }};
 handle_call({workspace_symbol, Query}, From, State) ->
-  #state{ request_id = RequestId
-        , socket     = Socket
-        } = State,
+  RequestId = State#state.request_id,
   Method = <<"workspace/symbol">>,
   Params = #{ query => Query },
   Content = erlang_ls_protocol:request(RequestId, Method, Params),
-  gen_tcp:send(Socket, Content),
+  send(Content, State),
   {noreply, State#state{ request_id = RequestId + 1
                        , pending    = [{RequestId, From} | State#state.pending]
                        }};
 handle_call({shutdown}, From, State) ->
-  #state{ request_id = RequestId
-        , socket     = Socket
-        } = State,
+  RequestId = State#state.request_id,
   Method = <<"shutdown">>,
   Params = #{},
   Content = erlang_ls_protocol:request(RequestId, Method, Params),
-  gen_tcp:send(Socket, Content),
+  send(Content, State),
   {noreply, State#state{ request_id = RequestId + 1
                        , pending    = [{RequestId, From} | State#state.pending]
                        }};
 handle_call({exit}, _From, State) ->
-  #state{ request_id = RequestId
-        , socket     = Socket
-        } = State,
+  RequestId = State#state.request_id,
   Method = <<"exit">>,
   Params = #{},
   Content = erlang_ls_protocol:request(RequestId, Method, Params),
-  gen_tcp:send(Socket, Content),
+  send(Content, State),
   {reply, ok, State}.
 
 -spec handle_cast(any(), state()) -> {noreply, state()}.
+handle_cast( {messages, Responses}
```

Comment:
This is handling the messages that `erlang_ls_stdio:loop/4` function sends throuhg `erlang_ls_server:send_requests/2`.